### ISSUE TITLE
Need help with migration of Liquigraph postconditions

### ISSUE THREAD

**Issue by shariqhaque** [liquibase/liquibase-neo4j#643]
We have historical changesets in liquigraph which uses postcondition to update in batches. We are in process oh upgrading to liquibase and, hence doing the migration. However, I am stuck in below issue:

1. Postcondition are not getting migrated and the cli tool gives an error. What's the workaround for this in a way that query isn't executed again after we migrate to Liquibase. Do we need a query change here?
2. We would like to make this a two step process, first in the code base we would like to change the liquigraph to liquibase xml and then at run time execute the database migration and do not produce the liquibase.xmls (Ofcourse it will be created and i can delete this but is there a way for avoid creaton of these liquibase.xmls?)

**Comment by fbiville** [comment]
Hello, 

Regarding your first question, you will likely need two steps:

- comment out the postcondition change set(s) and migrate your Liquigraph files and data to Liquibase
- then, migrate the postcondition change set(s) to queries that relies on `CALL {} IN TRANSACTIONS` or [`apoc.periodic.iterate`](https://neo4j.com/docs/apoc/current/graph-updates/periodic-execution/). 

For the latter to work, you'll need to explicit set `runInTransaction=false` to the enclosing change set(s) [as documented here](https://www.neo4j.com/labs/liquibase/docs/reference-features/#change-sets-runintransaction).

Regarding the second question, the best way forward would be to fork Liquigraph, add an boolean parameter to the `MigrateToLiquibase` and use it around the following piece of code: https://github.com/liquibase/liquigraph/blob/2ecef5c812319fae1b0713f7644d3c21d5e7ece1/liquigraph-cli/src/main/java/org/liquigraph/cli/commands/MigrateToLiquibase.java#L65-L75. That way, you can control whether you want migrate the Liquigraph data or not.

**Comment by shariqhaque** [comment]
@fbiville thanks a lot for the reply. 
1. I have already change the post considitions to apply apoc iterate, however this changes the checksum which leads to validation error. I also don't want these changeset to be executed again for internal reasons and hence using the validChecksum tags to make sure the changeset do not throw checksum error and at the same time if that's executed in a server, it shouldn't run again. 
2. I checked the internal implementation inside the liquigraph and now calling migrate persistence if liquibase doesn't exist in the startup of my sprinfboot.

Thanks a lot for providing such a smooth transition, I am not yet done with the migration and may ask further questions but will create diff ticket. Closing this issue
